### ISSUE TITLE
Openstack driver: Unclear instructions on overrides/configuration handling

### ISSUE THREAD

**Issue by johanssone** [ansible/molecule#1680]
# Issue Type

- Bug report

# Molecule and Ansible details

```
$ ansible --version && molecule --version
ansible 2.7.6
  config file = None
  configured module search path = [u'/home/ekan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ekan/dev/github/maas-venv/local/lib/python2.7/site-packages/ansible
  executable location = /home/ekan/dev/github/maas-venv/bin/ansible
  python version = 2.7.15rc1 (default, Nov 12 2018, 14:31:15) [GCC 7.3.0]
molecule, version 2.19.0

```

Molecule installation method (one of):
- pip

Ansible installation method (one of):
- pip

Detail any linters or test runners used:
- yamllint
- testinfra
# Desired Behavior
I find it somewhat unclear which ansible/molecule settings that should be overridable from `molecule.yml` for the Openstack driver. 
Example:
```
platforms:
  - name: maas-single-1604
    image: Ubuntu 16.04 Xenial Xerus
    flavor: 1C-1GB-50GB
    networks: Default network
    ssh_user: ubuntu
    sshuser: ubuntu
   user: ubuntu
    groups:
      - maas
```
Above configuration snippet properly overrides `image` and `flavor` also default set in `create.yml`. Overriding `networks` and the three different tests for `user` does not seem to work. 

I would expect common variables such as which network to use in OpenStack project, and which remote user to run tests as, to be overridable. Alternatively, a reference to values that are overridable and which are not.

Changing values for `neutron_network_name` and `ssh_user` directly in `create.yml` resolves the issue, but I'm not sure this is the intended behavior..
create.yml:
```
---
- name: Create
  hosts: localhost
  connection: local
  gather_facts: false
  no_log: "{{ not (lookup('env', 'MOLECULE_DEBUG') | bool or molecule_yml.provisioner.log|default(false) | bool) }}"
  vars:
    *ssh_user: ubuntu*
    ssh_port: 22
....
    *neutron_network_name: Default network*
```
# Actual Behaviour
See ^

**Comment by singuliere** [comment]
It is my understanding that the OpenStack create/destroy.yml provided by molecule via cookiecutter are meant as a starting point to improve rather than a fully functional way to provision hosts. For instance there is no way to describe volumes.

**Comment by singuliere** [comment]
Ideally the [molecule platforms](https://molecule.readthedocs.io/en/latest/configuration.html#platforms) would provide the same features as [HOT](https://docs.openstack.org/heat/latest/template_guide/hot_spec.html). But it would be redundant and a lot of work. An approach that would make sense is for create.yml to use a HOT templates instead. In the same spirit, the Docker driver should use [docker-compose.ym](https://docs.docker.com/compose/compose-file/). And the same applies for all providers for which there already exists a well documented way to describe the hosts, how they are networked, which storage is attached to them etc.

In other words, I think that trying to improve how `platforms` is interpreted for OpenStack is a never ending quest.

**Comment by lwm** [comment]
Molecule dogfoods Ansible all the way down. This is great because if you know Ansible, you know Molecule. So, I'd rather not move to formats that Ansible can't consume. 

I wonder if we can't completely avoid exposing the `create/prepare/destroy.yml` playbooks some time in the future and have the `molecule.yml` as the one and only entrypoint for configuration.

Looking forward to your PR @ragingpastry.

**Comment by singuliere** [comment]
> Molecule dogfoods Ansible all the way down. This is great because if you know Ansible, you know Molecule. So, I'd rather not move to formats that Ansible can't consume.

The [molecule platforms](https://molecule.readthedocs.io/en/latest/configuration.html#platforms) is a format that Ansible cannot consume. I'd rather deal with a well documented format that Ansible cannot consume than a home made format with little documentation because in the end it is more work, both for the implementor and the user.

**Comment by ssbarnea** [comment]
Openstack Driver was moved to https://github.com/pycontribs/molecule-openstack -- please re-add ticket there if still valid.
